datafusion/functions-aggregate-common/src/aggregate/groups_accumulator.rs

```diff
@@ -249,24 +303,92 @@ impl GroupsAccumulatorAdapter {
         let mut sizes_pre = 0;
         let mut sizes_post = 0;
         for (&group_idx, offsets) in iter {
-            let state = &mut self.states[group_idx];
-            sizes_pre += state.size();
-
-            let values_to_accumulate = slice_and_maybe_filter(
-                &values,
-                opt_filter.as_ref().map(|f| f.as_boolean()),
-                offsets,
-            )?;
-            f(state.accumulator.as_mut(), &values_to_accumulate)?;
+            sizes_pre += self.states[group_idx].size();
+            self.invoke_accumulator(group_idx, &values, offsets, opt_boolean_filter, f)?;
 
             // clear out the state so they are empty for next
             // iteration
-            state.indices.clear();
-            sizes_post += state.size();
+            self.states[group_idx].indices.clear();
+
+            sizes_post += self.states[group_idx].size();
         }
 
-        self.adjust_allocation(sizes_pre, sizes_post);
-        Ok(())
+        Ok((sizes_pre, sizes_post))
+    }
+
+    /// invokes f(accumulator, values) for each group that has values
+    /// in group_indices.
+    ///
+    /// This function is the same as [`Self::invoke_per_accumulator_on_non_ordered_group_indices`] but avoid reordering of the
+    /// input as we know that each group_index is contiguous
+    ///
```

Comment:
As above can you please document what the returned `()` tuple represents?

Reply:
updated